### PR TITLE
Jarvis 4 new vaults

### PR DESCRIPTION
**Pull request by Crypto-One-dev**

### CODE REVIEW

**Comment by Cryo45** [comment]
The png assets are too heavy: 200kb or more. Can you please reduce the resolution?

File: data/mainnet/addresses.json [hunk 1]
```diff
@@ -1778,6 +1786,30 @@
         "NewStrategy": "0xA87Abed957338A5dE2106D13d1967cF62d6d7996",
         "NewPool": "0xb42E1201cc041727B063cfbB506Ca7ee74B7857e"
       },
+      "jarvis_SES_2JPY": {
```
Comment (CryptJS13) [inline data/mainnet/addresses.json:1789]: I think all these addresses are already in this file, so no need to add them. Make sure you pull the latest version of `master` into your working branch.

File: data/mainnet/pools.js [hunk 2]
```diff
@@ -405,6 +405,126 @@ module.exports = [
...
+    rewardAPY: [],
+    rewardTokens: [addresses.MATIC.miFARM],
+    rewardTokenSymbols: ['miFARM', 'WMATIC'],
+    stakeAndDepositHelpMessage: `
+    <div class="help-message">
+      <b>Deposit and stake:</b>
+      <ol class="numeric-list">
+        <li>
+          Go to&nbsp;
+          <a target="_blank" rel="noopener noreferrer" href="https://dmm.exchange/#/add/0xAd326c253A84e9805559b73A08724e11E49ca651/0xf379CB529aE58E1A03E62d3e31565f4f7c1F2020/0x4924B6E1207EFb244433294619a5ADD08ACB3dfF">
```
Comment (CryptJS13) [inline data/mainnet/pools.js:429]: The link is pointing to the wrong pool. Please update with the correct link

File: data/mainnet/pools.js [hunk 3]
```diff
@@ -405,6 +405,126 @@ module.exports = [
...
+    rewardAPY: [],
+    rewardTokens: [addresses.MATIC.miFARM],
+    rewardTokenSymbols: ['miFARM', 'WMATIC', 'fSES_2JPY'],
+    stakeAndDepositHelpMessage: `
+    <div class="help-message">
+      <b>Deposit and stake:</b>
+      <ol class="numeric-list">
+        <li>
+          Go to&nbsp;
+          <a target="_blank" rel="noopener noreferrer" href="https://polygon.curve.fi/factory/37/deposit">
```
Comment (CryptJS13) [inline data/mainnet/pools.js:457]: Also wrong link

File: data/mainnet/pools.js [hunk 4]
```diff
@@ -405,6 +405,126 @@ module.exports = [
...
+    rewardAPY: [],
+    rewardTokens: [addresses.MATIC.miFARM],
+    rewardTokenSymbols: ['miFARM', 'WMATIC'],
+    stakeAndDepositHelpMessage: `
+    <div class="help-message">
+      <b>Deposit and stake:</b>
+      <ol class="numeric-list">
+        <li>
+          Go to&nbsp;
+          <a target="_blank" rel="noopener noreferrer" href="https://dmm.exchange/#/add/0xAd326c253A84e9805559b73A08724e11E49ca651/0xf379CB529aE58E1A03E62d3e31565f4f7c1F2020/0x4924B6E1207EFb244433294619a5ADD08ACB3dfF">
```
Comment (CryptJS13) [inline data/mainnet/pools.js:489]: Also wrong link

File: data/mainnet/pools.js [hunk 5]
```diff
@@ -405,6 +405,126 @@ module.exports = [
...
+            KyberDMM
+          </a>
+          and add liquidity for QUI_2CAD
```
Comment (CryptJS13) [inline data/mainnet/pools.js:492]: Use a normal dash (`-`) instead of the underscore (`_`) in QUI-2CAD

File: data/mainnet/pools.js [hunk 6]
```diff
@@ -405,6 +405,126 @@ module.exports = [
...
+    rewardAPY: [],
+    rewardTokens: [addresses.MATIC.miFARM],
+    rewardTokenSymbols: ['miFARM', 'WMATIC', 'fQUI_2CAD'],
+    stakeAndDepositHelpMessage: `
+    <div class="help-message">
+      <b>Deposit and stake:</b>
+      <ol class="numeric-list">
+        <li>
+          Go to&nbsp;
+          <a target="_blank" rel="noopener noreferrer" href="https://polygon.curve.fi/factory/37/deposit">
```
Comment (CryptJS13) [inline data/mainnet/pools.js:517]: Also wrong link

File: data/mainnet/tokens.js [hunk 7]
```diff
@@ -5413,6 +5445,21 @@ module.exports = {
...
   },
+  SES: {
+    chain: CHAINS_ID.MATIC_MAINNET,
+    logoUrl: '',
+    tokenAddress: addresses.MATIC.SES,
+    decimals: '18',
+    vaultAddress: null,
+    priceFunction: {
+      type: GET_PRICE_TYPES.KYBER_PAIR,
+      params: [
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5456]: The wrong parameters are in here. DEN should be SES

File: data/mainnet/tokens.js [hunk 8]
```diff
@@ -5575,6 +5644,84 @@ module.exports = {
...
+          addresses.MATIC.V2.jarvis_2JPY_HODL.RewardPool,
+          addresses.MATIC.V2.jarvis_2JPY_HODL.Underlying,
+          'jarvis_SES_2JPY',
+          profitSharingCut8Percent,
+        ],
+        extraDailyCompound: false,
+      },
+    ],
+    apyDescriptionOverride: [
+      'Auto harvested <b>SES</b> deposited into <b>SES(MAR22)-2JPY</b> for more APY',
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5682]: Just use SES, not SES(MAR22)

File: data/mainnet/tokens.js [hunk 9]
```diff
@@ -5575,6 +5644,84 @@ module.exports = {
...
+    ],
+    cmcRewardTokenSymbols: ['iFARM', 'SES'],
+  },
+  jarvis_2CAD_HODL: {
+    category: VAULT_CATEGORIES_IDS.JARVIS,
+    chain: CHAINS_ID.MATIC_MAINNET,
+    logoUrl: './icons/2jpy.png',
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5689]: Wrong logo

File: data/mainnet/tokens.js [hunk 10]
```diff
@@ -5575,6 +5644,84 @@ module.exports = {
...
+        ['JCAD', 'CADC'],
+        CHAINS_ID.MATIC_MAINNET,
+      ],
+    },
+    estimateApyFunctions: [
+      {
+        type: 'JARVIS_HODL_V2',
+        params: [
+          0,
+          addresses.MATIC.V2.jarvis_2CAD_HODL.RewardPool,
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5712]: Address is not defined

File: data/mainnet/tokens.js [hunk 11]
```diff
@@ -5575,6 +5644,84 @@ module.exports = {
...
+          addresses.MATIC.V2.jarvis_2CAD_HODL.Underlying,
+          'jarvis_SES_2JPY',
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5714]: Wrong hodl vault

File: data/mainnet/tokens.js [hunk 12]
```diff
@@ -5575,6 +5644,84 @@ module.exports = {
...
+          profitSharingCut8Percent,
+        ],
+        extraDailyCompound: false,
+      },
+    ],
+    apyDescriptionOverride: [
+      'Auto harvested <b>SES</b> deposited into <b>SES(MAR22)-2JPY</b> for more APY',
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5721]: Wrong text, should be QUI and 2CAD

File: data/mainnet/tokens.js [hunk 13]
```diff
@@ -5575,6 +5644,84 @@ module.exports = {
...
+    ],
+    cmcRewardTokenSymbols: ['iFARM', 'SES'],
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5723]: Should be QUI

File: data/mainnet/tokens.js [hunk 14]
```diff
@@ -5632,6 +5779,56 @@ module.exports = {
     ],
     cmcRewardTokenSymbols: ['iFARM', 'DEN'],
   },
+  jarvis_SES_2JPY: {
+    category: VAULT_CATEGORIES_IDS.INACTIVE_POLYGON,
+    inactive: true,
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5784]: Wrong category, should not be inactive

File: data/mainnet/tokens.js [hunk 15]
```diff
@@ -5632,6 +5779,56 @@ module.exports = {
...
+    chain: CHAINS_ID.MATIC_MAINNET,
+    logoUrl: './icons/ses.png',
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5786]: Wrong logo

File: data/mainnet/tokens.js [hunk 16]
```diff
@@ -5632,6 +5779,56 @@ module.exports = {
...
+    displayName: 'Jarvis: SES-2JPY',
+    subLabel: 'Auto-compounding',
+    tokenAddress: addresses.MATIC.V2.jarvis_SES_2JPY.Underlying,
+    decimals: '18',
+    vaultAddress: addresses.MATIC.V2.jarvis_SES_2JPY.NewVault,
+    priceFunction: {
+      type: GET_PRICE_TYPES.LP_TOKEN,
+      params: [addresses.MATIC.V2.jarvis_SES_2JPY.Underlying, 'SES', 'jarvis_2JPY_HODL'],
+    },
+    estimateApyFunctions: [
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5798]: Use regular JARVIS_V2 apy calculation here. This is not a hodl-vault. These are the input parameters for JARVIS_V2: `poolId, rewardPool, underlying, reduction`

File: data/mainnet/tokens.js [hunk 17]
```diff
@@ -5632,6 +5779,56 @@ module.exports = {
...
+      {
+        type: 'JARVIS_HODL',
+        params: [0, addresses.MATIC.V2.jarvis_SES_2JPY.Underlying, profitSharingCut8Percent],
+        extraDailyCompound: false,
+      },
+    ],
+    cmcRewardTokenSymbols: ['iFARM', 'SES'],
+  },
+  jarvis_QUI_2CAD: {
+    category: VAULT_CATEGORIES_IDS.INACTIVE_POLYGON,
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5808]: Wrong category

File: data/mainnet/tokens.js [hunk 18]
```diff
@@ -5632,6 +5779,56 @@ module.exports = {
...
+    inactive: true,
+    chain: CHAINS_ID.MATIC_MAINNET,
+    logoUrl: './icons/qui.png',
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5811]: Wrong icon

File: data/mainnet/tokens.js [hunk 19]
```diff
@@ -5632,6 +5779,56 @@ module.exports = {
...
+    displayName: 'Jarvis: QUI-2CAD',
+    subLabel: 'Auto-compounding',
+    tokenAddress: addresses.MATIC.V2.jarvis_QUI_2CAD.Underlying,
+    decimals: '18',
+    vaultAddress: addresses.MATIC.V2.jarvis_QUI_2CAD.NewVault,
+    priceFunction: {
+      type: GET_PRICE_TYPES.LP_TOKEN,
+      params: [addresses.MATIC.V2.jarvis_QUI_2CAD.Underlying, 'QUI', 'jarvis_2CAD_HODL'],
+    },
+    estimateApyFunctions: [
```
Comment (CryptJS13) [inline data/mainnet/tokens.js:5823]: Use JARVIS_V2

**Review by CryptJS13 — CHANGES_REQUESTED** [review]
Quite some comments. Also a lot of them seem small copy/paste errors. Please make sure you work neatly and keep an eye on the details. Also make sure to test your outputs. I am sure this code would not have produced the results you expected.

Some more general points:
- Please put new additions to `pools.js` all the way on the top of the file.
- Please put new addition to `tokens.js` to the top of the file, below `WETH`.
- Please make sure the vaults are marked with the `isNew` tag on release

**Comment by Crypto-One-dev** [comment]
> The png assets are too heavy: 200kb or more. Can you please reduce the resolution?

I see other images bigger than 200kb, should I work on them also?

File: data/mainnet/tokens.js (hunk 11 above)
Comment (Crypto-One-dev) [reply data/mainnet/tokens.js:5714]: > Wrong hodl vault

Could you tell me what's wrong here please?

File: data/mainnet/tokens.js [hunk 20]
```diff
@@ -5575,6 +5644,84 @@ module.exports = {
...
+        ['JJPY', 'JPYC'],
+        CHAINS_ID.MATIC_MAINNET,
+      ],
+    },
+    estimateApyFunctions: [
+      {
+        type: 'JARVIS_HODL_V2',
+        params: [
+          0,
+          addresses.MATIC.V2.jarvis_2JPY_HODL.RewardPool,
```
[the remark this answers is not in the document]
Comment (Crypto-One-dev) [reply data/mainnet/tokens.js:5673]: > This address is not currently defined

How can I update this?
I just followed other vault sturcture

[the remark this answers is not in the document]
Comment (CryptJS13) [reply data/mainnet/tokens.js:5673]: Add the address to the `addresses.json`

File: data/mainnet/tokens.js (hunk 11 above)
Comment (CryptJS13) [reply data/mainnet/tokens.js:5714]: It should be jarvis_QUI_2CAD

**Review by CryptJS13 — COMMENTED** [review]
Looks good, except for the ordering. Also not sure about the icons. Seems like there are some duplicates, and not all of them are being used.

**Comment by CryptJS13** [comment]
@Crypto-One-dev Please make sure the conflicts are resolved

@Cryo45 After that is done this should be good to merge

**Pull request closed without merging**